### DISCUSSION TITLE
exploring-dotnet-6-part-12-upgrading-a-dotnet-5-startup-based-app-to-dotnet-6/

### DISCUSSION THREAD

**Discussion by giscus** [andrewlock/blog-comments#77] [Announcements]
# exploring-dotnet-6-part-12-upgrading-a-dotnet-5-startup-based-app-to-dotnet-6/

In this post I tackle a question I have received several times: how can I update an ASP.NET Core 5 app that uses Startup to .NET 6's minimal hosting APIs.

https://andrewlock.net/exploring-dotnet-6-part-12-upgrading-a-dotnet-5-startup-based-app-to-dotnet-6/

**Comment by MikeMcWilliams** [comment]
This is really good article to help re-use the Startup class.  The only caveat that I see is that startup.Configure may use dependency injection.  What do you suppose we get around that (because the ```new Startup``` suggested doesn't seem to fly.)

```public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<Startup> logger)```

**Reply by falvarez1** [reply]
It's a bit of a rabbit hole but you can start by seeing how the AspNetCore team does their magic:
https://github.com/dotnet/aspnetcore/blob/main/src/Hosting/Hosting/src/GenericHost/GenericWebHostBuilder.cs#L322-L344

**Reply by andrewlock** [reply]
> The only caveat that I see is that startup.Configure may use dependency injection. What do you suppose we get around that (because the new Startup suggested doesn't seem to fly.)

For your specific example, I would simply call your `Startup.Configure()` method like this:

```csharp
var builder = WebApplication.CreateBuilder(args);

var startup = new Startup(builder.Configuration);
startup.ConfigureServices(builder.Services);

var app = builder.Build();

startup.Configure(app, app.Environment, app.Logger);
app.Run();

public class Startup
{
    public void ConfigureServices(IServiceCollection services) { } 
    public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger logger)
}
```

More generally, I would grab the required services from the container and inject them manually:

```csharp
var builder = WebApplication.CreateBuilder(args);

var startup = new Startup(builder.Configuration);
startup.ConfigureServices(builder.Services);

var app = builder.Build();

var someService = app.Services.GetRequiredService<SomeService>(); // 👈 grab the service from DI
startup.Configure(app, someService); // 👈 Inject it
app.Run();

public class Startup
{
    public void ConfigureServices(IServiceCollection services) { } 
    public void Configure(IApplicationBuilder app, SomeService service) { }
}
```

**Comment by caseyspaulding** [comment]
Thanks going to try this out

[a comment by an automated account is not shown]
